plugins/modules/azure_rm_backuppolicy.py

```diff
@@ -255,28 +318,50 @@ def exec_module(self, **kwargs):
                     return self.results
 
                 response = self.create_or_update_backup_policy()
-                self.set_results(response)
+                self.results.update(self.set_results(response))
 
             # log that we're doing an update
             else:
                 self.log("Backup policy {0} for vault {1} in resource group {2} already exists, updating".format(self.name,
                                                                                                                  self.vault_name,
                                                                                                                  self.resource_group))
-
-                self.results['changed'] = True
+                if self.schedule_run_frequency is not None and self.schedule_run_frequency != old_res['properties']['schedule_policy']['schedule_run_frequency']:
+                    self.results['changed'] = True
+                elif self.time_zone is not None and self.time_zone != old_res['properties']['time_zone']:
+                    self.results['changed'] = True
+                elif self.instant_recovery_snapshot_retention is not None and self.instant_recovery_snapshot_retention != old_res['properties']['instant_rp_retention_range_in_days']:
+                    self.results['changed'] = True
+                elif self.schedule_weekly_frequency is not None and self.schedule_weekly_frequency != old_res['properties']['schedule_policy']['schedule_weekly_frequency']:
+                    self.results['changed'] = True
+                elif self.weekly_retention_count is not None and self.weekly_retention_count != old_res['properties']['retention_policy'].get('weekly_schedule', {}).get('retention_duration', {}).get('count'):
```

Comment:
line too long, over than 160

Reply:
addressed via rebase on top of your commits in #1890